homeassistant/components/home_connect/sensor.py

```diff
@@ -242,6 +298,198 @@ class HomeConnectSensorEntityDescription(
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="descaling_in_20_cups",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DESCALING_IN_15_CUPS,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="descaling_in_15_cups",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DESCALING_IN_10_CUPS,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="descaling_in_10_cups",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DESCALING_IN_5_CUPS,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="descaling_in_5_cups",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DEVICE_SHOULD_BE_DESCALED,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="device_should_be_descaled",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DEVICE_DESCALING_OVERDUE,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="device_descaling_overdue",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DEVICE_DESCALING_BLOCKAGE,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="device_descaling_blockage",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DEVICE_SHOULD_BE_CLEANED,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="device_should_be_cleaned",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_DEVICE_CLEANING_OVERDUE,
+        device_class=SensorDeviceClass.ENUM,
+        options=EVENT_OPTIONS,
+        default_value="off",
+        translation_key="device_cleaning_overdue",
+        appliance_types=("CoffeeMaker",),
+    ),
+    HomeConnectSensorEntityDescription(
+        key=EventKey.CONSUMER_PRODUCTS_COFFEE_MAKER_EVENT_CALC_N_CLEAN_IN20CUPS,
```

Comment:
Side note: I wonder why we didn't put underscores between "in 20 cups" here, but we did it for descaling.